compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -547,6 +534,25 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
         Ok(())
     }
 
+    fn static_mutability(&self, mutability: Mutability, nested: bool, did: DefId) -> Mutability {
+        match (mutability, nested) {
+            (Mutability::Mut, _) => Mutability::Mut,
+            (Mutability::Not, true) => Mutability::Not,
+            (Mutability::Not, false)
+                if !self
+                    .ecx
+                    .tcx
+                    .type_of(did)
+                    .no_bound_vars()
+                    .expect("statics should not have generic parameters")
+                    .is_freeze(*self.ecx.tcx, ty::ParamEnv::reveal_all()) =>
```

Comment:
I know this is pre-existing, but... why is `reveal_all` okay here? Shouldn't this be `ecx.param_env()`?

Reply:
We're asking for information about a static item's type. If anything we should get the param env of the static (but we know that is just `ParamEnv::reveal_all()` anyway.